lib/cppapi/AsyncTimer.cc

```diff
@@ -93,11 +93,17 @@ AsyncTimer::run()
 void
 AsyncTimer::cancel()
 {
-  if (!state_->cont_) {
+  // Assume this object is locked and the state isn't being updated elsewhere.
+  // Note that is not the same as the contained continuation being locked.
+  auto contp{state_->cont_}; // save this
```

Comment:
How does the compiler know the type of contp?  Some explicitness would be good here.